src/lib/format/tests/TestDecoding.cpp

```diff
@@ -130,48 +128,48 @@ void TestSampleData(nlTestSuite * inSuite, const PayloadDecoderInitParams & para
         printf("ACTUAL: '%s'\n", partial.Reset().Add(output_builder.c_str() + idx).AddMarkerIfOverflow().c_str());
     }
 
-    NL_TEST_ASSERT(inSuite, strcmp(output_builder.c_str(), expectation) == 0);
+    EXPECT_EQ(strcmp(output_builder.c_str(), expectation), 0);
```

Comment:
Please use string-aware comparisons for strcmp (I think they would show better messages in case of mismatch):

https://pigweed.dev/pw_unit_test/#c.EXPECT_STREQ

Reply:
Done